docs/_posts/2017-11-30-nullannotations.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: "Less NPE's - We now use Eclipse Nullannotations!"
+date:   2017-11-30
+image: "2017-11-30-nullannotations.png"
+published: true
+---
+
+Nullpointer Exceptions (aka NPE's) are pretty nasty since they are `RuntimeException`s and thus developers do not immediately notice them while coding in their IDE.
+Hence, data-flow analysis where written that show potential accesses to nullpointers before they occur at runtime. Our project decided to make use of these ananlyses and get rid of (some) nullpointers.
+
+<!--more-->
+
+Two popular projects that offer such an analysis are the [Checker Framework](https://checkerframework.org) and the [Eclipse JDT compiler](https://wiki.eclipse.org/JDT_Core/Null_Analysis).
+In order to obtain better results from these anylyses, developers can make use of `annotations` in the code.
+In the current snapshots and in our next release we support such annotations in our provided Eclipse IDE setup and maven configuration.
+We decided to use the implementation offered by the Eclipse JDT compiler.
+
+#### Making JavaDoc comments explicit
+
+Since we are a framework that will be integrated into solutions which are build by others, we have provided proper Javadoc documentation on our public interfaces in the past.
+In these Javadoc comments we state that a method will never return `null`, a parameter should never be `null` or is optional and thus maybe `null`.
+Maintaining these comments by hand is error-prone and subject to be ignored by developers not reading them.
+Therefore we have decided to provide these comments in a machine readable way so the IDE can support developers in showing them errors or warnings in case they violate them.
+
+As described in our [Coding Guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style) we annotate classes with the `@NonNullByDefault` annotations which declares that every field, return value, parameter, etc. defined in this class will always have a value unequal to `null`.
```

Comment:
* As described in our [Coding Guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style)**,** [...]

* What is "etc."? Isn't this list complete already?

* "defined in this class" -> "defined in such classes"

* "will always have a value unequal to `null`" "will always have value**s** other than `null`" (or "...not equals to `null`)